### PR TITLE
Fix #16432 - openbsd fork+attach EBUSY issue ##debug

### LINKED ISSUE #16432: Debugger broken on OpenBSD.
### Work environment

 * OpenBSD-current amd64
 * ELF

```
radare2 4.4.0-git 24204 @ openbsd-x86-64 git.4.3.1-206-g38b8d5929
commit: 38b8d5929d1855e7ca6b5b82956c03b46dd07eef build: 2020-04-05__22:02:03
```

### Expected behavior

```
$ r2 -d /tmp/a.out
```

Should open debug session for the binary.

### Actual behavior

```
Process with PID 15528 started...
Process with PID 59532 started...
[w] Cannot open 'dbg:///tmp/a.out' for writing.
```

and exits.

Also tried `ood`:

```
$ r2 /tmp/a.out   
 -- Setup dbg.fpregs to true to visualize the fpu registers in the debugger view.
[0x00001000]> ood
Process with PID 62226 started...
Process with PID 35393 started...
r_core_file_reopen: Cannot reopen file: dbg:///tmp/a.out  with perms 0x7, attempting to open read-only.
[0x00000000]> db main
Cannot place a breakpoint on 0x00001340 unmapped memory.See e? dbg.bpinmaps
```

Also found this seg:
```
$ r2 -                        
 -- This is fine.
[0x00000000]> ood /tmp/a.out
Cannot debug file (malloc://512) with permissions set to 0x7.
Reopening the original file in read-only mode.
zsh: segmentation fault (core dumped)  r2 -
```

All of these behaviours seem to work on Debian, so I assume this is OpenBSD specific.

Thanks

### PR DESCRIPTION
**Pull request by radare**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**
see the issue

**Test plan**

we dont test debugger in bsd targets

**Closing issues**

fix #16432

### CODE REVIEW

**Comment by radare** [comment]
It is assuming ebusy error as already attached

**Comment by ret2libc** [comment]
> It is assuming ebusy error as already attached

Oh, I see now you have removed the final `return NULL;`. Nice!

**Comment by thestr4ng3r** [comment]
I pushed a better fix. `EBUSY` IS an error normally so it's better to not ignore it.

File: libr/io/p/io_ptrace.c [hunk 1]
```diff
@@ -178,14 +178,18 @@ static bool __plugin_open(RIO *io, const char *file, bool many) {
...
 	siginfo_t *sig = (siginfo_t *)data;
 	return -1 != r_io_ptrace (io, PTRACE_GETSIGINFO, pid, NULL, sig);
 #elif defined(__FreeBSD__)
 	struct ptrace_lwpinfo *info = (struct ptrace_lwpinfo *)data;
 	int len = (int)sizeof (*info);
 	return -1 != r_io_ptrace (io, PT_LWPINFO, pid, info, len);
+#elif __OpenBSD__
+	ptrace_state_t *state = data;
+	int len = (int)sizeof (*state);
+	return -1 != r_io_ptrace (io, PT_GET_PROCESS_STATE, pid, state, len);
```
Comment (radare) [inline libr/io/p/io_ptrace.c:192]: We use to put thisb-1 at the right of the =

File: libr/io/p/io_ptrace.c [hunk 2]
```diff
@@ -220,14 +226,15 @@ static RIODesc *__open(RIO *io, const char *file, int rw, int mode) {
...
 				ret = pid;
 				eprintf ("ptrace_attach: Operation not permitted\n");
-				break;
+				return NULL;
 			case EINVAL:
 				perror ("ptrace: Cannot attach");
 				eprintf ("ERRNO: %d (EINVAL)\n", errno);
-				break;
+				return NULL;
+			default:
```
Comment (radare) [inline libr/io/p/io_ptrace.c:234]: Just return null once if ebusy is not ignores now.

Also. Is your patch working? Did u tested it?

Comment (thestr4ng3r) [reply libr/io/p/io_ptrace.c:234]: > Just return null once if ebusy is not ignores now.
wat

> Also. Is your patch working? Did u tested it?
yes

**Pull request merged**